lib/internal/timers.js

```diff
@@ -394,8 +394,7 @@ function compareTimersLists(a, b) {
   if (expiryDiff === 0) {
     if (a.id < b.id)
       return -1;
-    if (a.id > b.id)
-      return 1;
```

Comment:
if `a.id` equal to `b.id` return `expiryDiff = 0`, and the `childIndex` won't be changed. but this new logic has some differences, maybe.

https://github.com/nodejs/node/blob/eac3f0adc46c3af0e94d70bb8635440382df7d7f/lib/internal/priority_queue.js#L58

I definitely won't change this.

-1

Reply:
As I mentioned, there cannot be a case when the id of the two nodes can be equal. Can you provide a scenario when there is a comparison of two matching nodes as if it is done then there is probably a fault in the implementation of `priority_queue` itself? Not convinced around the case for the same id.

Reply:
Also, insertion is only done at one place. So nothing to worry for the `id` check.

https://github.com/nodejs/node/blob/da9c1ca8d3af525862fbf56755fd481f883aca19/lib/internal/timers.js#L350-L351